src/Microsoft.AspNetCore.Routing/Template/TemplateBinder.cs

```diff
@@ -87,12 +127,12 @@ public TemplateValuesResult GetValues(RouteValueDictionary ambientValues, RouteV
                     }
```

Comment:
I'm not convinced this check is doing the right thing with your changes, it would need to to remove things from the accepted values if they can't be used.